httpx/middleware/timeout.go

```diff
@@ -95,17 +103,20 @@ func (h *timeoutHandler) ServeHTTPContext(ctx context.Context, w http.ResponseWr
 }
 
 type timeoutWriter struct {
-	w    http.ResponseWriter
 	h    http.Header
 	wbuf bytes.Buffer
 
 	mu          sync.Mutex
 	timedOut    bool
 	wroteHeader bool
+	modified    bool
 	code        int
 }
 
-func (tw *timeoutWriter) Header() http.Header { return tw.h }
+func (tw *timeoutWriter) Header() http.Header {
+	tw.modified = true
```

Comment:
why is `modified` set to `true` on a getter method?

also, in case it is really necessary, should this method acquire `tw.mu.Lock()`?

Reply:
Its common to call `tw.Header().Set("Key", "Val")`. Good point about lock.

Reply:
I see. Do you want to write an inline comment to explain it? it seems non-intuitive to me.

Reply:
Ok, I changed it so we just look to see if there are any headers set or not